libbeat/cmd/instance/beat.go

```diff
@@ -273,11 +273,14 @@ func (b *Beat) launch(bt beat.Creator) error {
 	}
 
 	if b.Config.Monitoring.Enabled() {
-		reporter, err := report.New(b.Info, b.Config.Monitoring, b.Config.Output)
-		if err != nil {
-			return err
+		reportingActive := b.checkXPackMonitoring()
```

Comment:
It seems like this logic is integral to the Elasticsearch monitoring output. So why not have all of the related code to check if x-pack monitoring is enabled  part of the same package?

Reply:
Looking into this I realised we already have an almost identical code here: https://github.com/elastic/beats/blob/master/libbeat/monitoring/report/elasticsearch/client.go#L41 I need to revisit the implementation to see how the two can be combined.